### ISSUE TITLE
[FEATURE] STAR handle gzipped files

### ISSUE THREAD

**Issue by Emiller88** [nf-core/modules#1688]
### Is your feature request related to a problem? Please describe

I'd be cool if we didn't have to include

```nextflow
    withName: STAR_ALIGN {
        ext.args = '--readFilesCommand zcat'
    }
```

### Describe the solution you'd like

```nextflow
    def gzip = reads.name.endsWith(".gz") ? "--readFilesCommand zcat"
    """
    STAR \\
        --genomeDir $index \\
        --readFilesIn $reads  \\
        --runThreadN $task.cpus \\
        --outFileNamePrefix $prefix. \\
        $out_sam_type \\
        $ignore_gtf \\
        $seq_center \\
        $gzip \\
        $args
```
I can't remember if `:` is necessary or not in the Elvis operator.

### Describe alternatives you've considered

- Not doing this and leaving it up to users to figure it out.
- We might want to handle overriding that with `args2` maybe or some other method that checks the `ext.args` for a `--readFilesCommand`

**Comment by grst** [comment]
How comes we don't have this already? This must be handled in the rnaseq pipeline somehow? 

`--readFilesCommand gzip -cdf` should work with both compressed and uncompressed files, making the groovy conditional superflous.

**Comment by Emiller88** [comment]
Exactly. Which works, but it's a foot gun, when we can just use the power of Nextflow to handle STAR not doing the job and handling it, or maybe just giving users too many options rather than telling them what they want. But we can fix it on our end.

**Comment by grst** [comment]
> We might want to handle overriding that with args2 maybe or some other method that checks the ext.args for a --readFilesCommand

How about: 
 * check if `--readFilesCommand` is in `args`
 * if not, add `--readFilesCommand gzip -cdf`

This would also be backwards-compatible with current implementations (e.g. rnaseq) and provides sensible defaults to work with compressed files out-of-the-box.

**Comment by ramirobarrantes** [comment]
I am trying to fix this, but I am stuck on this line:

`def gzip = reads.getName().endsWith(".gz") ? "--readFilesCommand zcat" : ''`

   Nextflow stdout:                                                                                       │
│                                                                                                          │
│   ERROR ~ Module compilation error                                                                       │
│   - file :                                                                                               │
│ ...                                                   │
│   - cause: Variable `reads` already defined in the process scope @ line 51, column 16.                   │
│          def gzip = reads.getName().endsWith(".gz") ? "--readFilesCommand zcat" :

**Comment by edmundmiller** [comment]
@ramirobarrantes could you link me to your updates to the module? I'm struggling without the rest of the module to know why that line is throwing an error 😅

**Comment by ramirobarrantes** [comment]
> @ramirobarrantes could you link me to your updates to the module? I'm struggling without the rest of the module to know why that line is throwing an error 😅

Do you mean that you want me to push the broken changes? Or just the location of where the changes should be? I haven't updated anything as they don't work yet.

**Comment by edmundmiller** [comment]
Sure, just push up the broken changes you have!

**Comment by ramirobarrantes** [comment]
Done...it was just on nextflow.config and on main.nf, it's probably very simple fix.

**Comment by famosab** [comment]
@edmundmiller @ramirobarrantes Is this still an opn issue?

**Comment by vagkaratzas** [comment]
Can't find the relevant PR, and this still seems to need the update. Anyone working on it?

`star_starsolo` seems to have already implemented this: `def zcat = reads[0].getExtension() == "gz" ? "--readFilesCommand zcat": ""`
